src/main/java/thewaypointers/trafficsimulator/gui/PanelMouseAction.java

```diff
@@ -0,0 +1,131 @@
+package thewaypointers.trafficsimulator.gui;
+
+import thewaypointers.trafficsimulator.common.JunctionDTO;
+import thewaypointers.trafficsimulator.common.TrafficLightSystemDTO;
+
+import javax.swing.*;
+import javax.swing.event.MouseInputListener;
+import java.awt.*;
+import java.awt.event.MouseEvent;
+import java.util.*;
+import java.util.List;
+
+public class PanelMouseAction implements MouseInputListener {
+    Point origin;
+    private JPanel panel;
+    private int panel_X;
+    private int panel_Y;
+
+    public PanelMouseAction(JPanel panel) {
+
+        this.panel = panel;
+        origin = new Point();
+        panel_X = panel.getWidth();
+        panel_Y = panel.getHeight();
+    }
+
+    @Override
+    public void mouseClicked(MouseEvent e) {
+
+        Point p = e.getPoint();
+        double x = p.x;
+        double y = p.y;
+        //System.out.println(x+", "+y);
+
+        TrafficLightSystemDTO trafficLightSystemDTO = MainFrame.mapContainerPanel.mapPanel.worldState.getTrafficLightSystem();
+        List<String> labels = trafficLightSystemDTO.GetJunctionLabel();
+
+        //System.out.println(labels.size());
+
+        for (String label : labels) {
+            double junction_x = MainFrame.mapContainerPanel.mapPanel.GetJunctionLocation(label).getX();
+
+            double junction_y = MainFrame.mapContainerPanel.mapPanel.GetJunctionLocation(label).getY();
+
+            if (((junction_x <= x) && (x <= junction_x + 50)) && ((junction_y <= y) && (y <= junction_y + 50))) {
+                //pass label here
+                System.out.println("click test, pass label to the simulation");
```

Comment:
It looks good, but could you make it so it actually passes the information somewhere? For example it could call the method `SimulationParameterChanged` on an `ISimulationInputListener` object (we have that interface already). That way when someone will be implementing this on the simulation side, we will be able to use it easily and not have to change code here.
.